### PR TITLE
feat: adds disablePictureInPicture method to the player API.

### PR DESCRIPTION
**Pull request by gjanblaszczyk**
## Description
A proposed fix for #6341 

## Specific Changes proposed
Add `disablePictureInPicture` method the the player API.

## Requirements Checklist
- [X] Feature implemented / Bug fixed
- [X] If necessary, more likely in a feature request than a bug fix
  - [X] Change has been verified in an actual browser (Chome, Firefox, IE)
  - [X] Unit Tests updated or fixed
  - [ ] Docs/guides updated
  - [ ] Example created ([starter template on JSBin](https://codepen.io/gkatsev/pen/GwZegv?editors=1000#0))
- [ ] Reviewed by Two Core Contributors

### CODE REVIEW

**Comment by gjanblaszczyk** [comment]
This PR is also related to #5824

**Comment by kevinfaguiar** [comment]
Any news on this?

**Comment by gjanblaszczyk** [comment]
I'm still waiting for review. I am pretty sure that @gkatsev will do the review sooner or later but it is hard to tell when this will happen exactly...

File: src/js/player.js [hunk 1]
```diff
@@ -2852,6 +2853,26 @@ class Player extends Component {
...
+  disablePictureInPicture(value) {
+    if (value === undefined) {
+      return this.techGet_('disablePictureInPicture');
+    }
+    this.techCall_('setDisablePictureInPicture', value);
+    this.options_.disablePictureInPicture = value;
+    const PiPToggle = this.controlBar.pictureInPictureToggle;
+
+    if (PiPToggle) {
+      PiPToggle.handlePiPEnabledChange();
```
Comment (gkatsev) [inline src/js/player.js]: what about triggering something like `disablepictureinpicturechanged` and having the PiPToggle listen to that in addition to loadedmetadata? That way we don't need to worry about getting the toggle directly and someone could use that for a custom pip toggle as well?

File: src/js/control-bar/picture-in-picture-toggle.js [hunk 2]
```diff
@@ -27,13 +27,10 @@ class PictureInPictureToggle extends Button {
   constructor(player, options) {
     super(player, options);
     this.on(player, ['enterpictureinpicture', 'leavepictureinpicture'], this.handlePictureInPictureChange);
+    this.on(player, 'loadedmetadata', this.handlePiPEnabledChange);
```
Comment (gkatsev) [inline src/js/control-bar/picture-in-picture-toggle.js]: could listen here to `disablepictureinpicturechanged` (a custom event) as outlined in the player method.

File: src/js/tech/html5.js [hunk 3]
```diff
@@ -1747,7 +1764,8 @@ Html5.resetMediaElement = function(el) {
 // Wrap native properties with a setter in this format:
 // set + toTitleCase(name)
 // The list is as follows:
-// setVolume, setSrc, setPoster, setPreload, setPlaybackRate, setDefaultPlaybackRate
+// setVolume, setSrc, setPoster, setPreload, setPlaybackRate, setDefaultPlaybackRate,
+// setDisablePictureInPicture
```
Comment (gkatsev) [inline src/js/tech/html5.js]: thanks for adding that here. Makes grepping through the code possible.

File: test/api/api.js [hunk 4]
```diff
@@ -39,6 +39,7 @@ QUnit.test('should be able to access expected player API methods', function(asse
   assert.ok(player.textTracks, 'textTracks exists');
   assert.ok(player.requestFullscreen, 'requestFullscreen exists');
   assert.ok(player.exitFullscreen, 'exitFullscreen exists');
+  assert.ok(player.disablePictureInPicture, 'disablePictureInPicture exists');
```
Comment (gkatsev) [inline test/api/api.js]: thanks, though, I wonder if these tests even run anymore... 🤔

**Comment by gkatsev** [comment]
Thanks for your patience, needed to take some time away from answering issues and reviewing PRs. Back at it, at least in some capacity now.

File: src/js/player.js (hunk 1 above)
Comment (gjanblaszczyk) [reply src/js/player.js]: Nice idea. I have just updated the PR.

File: src/js/control-bar/picture-in-picture-toggle.js (hunk 2 above)
Comment (gjanblaszczyk) [reply src/js/control-bar/picture-in-picture-toggle.js]: Thx, I added support for this event.

**Comment by gjanblaszczyk** [comment]
Thanks @gkatsev for the review.  I have just added changes according to your comments.

File: src/js/control-bar/picture-in-picture-toggle.js [hunk 5]
```diff
@@ -46,6 +43,18 @@ class PictureInPictureToggle extends Button {
     return `vjs-picture-in-picture-control ${super.buildCSSClass()}`;
   }
 
+  /**
+   * Enables or disables button based on document.pictureInPictureEnabled property value
+   * or on value returned by player.disablePictureInPicture() method.
+   */
+  handlePiPEnabledChange() {
```
Comment (brandonocasey) [inline src/js/control-bar/picture-in-picture-toggle.js:50]: What do you think of calling this `handlePictureInPictureEnabledChange` for consistency. I know it's really long, but it seems weird to only use `PiP` here when we use `PictureInPicture` everywhere else.

Comment (gjanblaszczyk) [reply src/js/control-bar/picture-in-picture-toggle.js]: Hi, I don't have a problem with that. I have just changed the method name to fit your expectations.

**Pull request merged**